src/store/juju/slice.ts

```diff
@@ -243,18 +243,16 @@ const slice = createSlice({
     },
     updateCrossModelQuery: (
       state,
-      { payload }: PayloadAction<CrossModelQueryFullResponse>,
+      { payload }: PayloadAction<CrossModelQueryResponse | Error>,
     ) => {
-      // If "payload" is a string, it represents the error. In this case,
-      // "results" gets set to null and "errors" gets set to "payload".
       state.crossModelQuery.results = isCrossModelQueryResponse(payload)
         ? payload.results
         : null;
       state.crossModelQuery.errors =
         isCrossModelQueryResponse(payload) && Object.keys(payload.errors).length
           ? payload.errors
-          : typeof payload === "string"
-            ? payload
+          : payload instanceof Error
+            ? payload.message
```

Comment:
```suggestion
          : payload instanceof Error
            ? payload.message
```
      const hasErrors = Object.keys(payload.errors).length > 0;
      state.crossModelQuery.results = hasErrors ? null : payload.results;
      state.crossModelQuery.errors = hasErrors ? payload.errors : null;